pallets/xdns/src/lib.rs

```diff
@@ -432,21 +375,32 @@ pub mod pallet {
             )
         }
 
-        fn get_gateway_security_coordinates(chain_id: &ChainId) -> Result<Bytes, &'static str> {
+        /// returns the gateway vendor of a gateway if its available
+        fn get_gateway_vendor(chain_id: &ChainId) -> Result<GatewayVendor, DispatchError> {
+            match <XDNSRegistry<T>>::get(&chain_id) {
+                Some(rec) => Ok(rec.gateway_vendor),
+                None => Err(Error::<T>::XdnsRecordNotFound.into()),
+            }
+        }
+
+        fn get_gateway_security_coordinates(chain_id: &ChainId) -> Result<Bytes, DispatchError> {
             if !<XDNSRegistry<T>>::contains_key(chain_id) {
-                return Err("Xdns record not found while accessing security coordinates")
+                return Err(Error::<T>::XdnsRecordNotFound.into())
             }
             Ok(<XDNSRegistry<T>>::get(chain_id)
-                .ok_or("XDNSRegistry does not contain given chain id")?
-                .security_coordinates)
+                .unwrap()
+                .security_coordinates) //safe because checked
         }
 
-        fn get_gateway_para_id(chain_id: &ChainId) -> Result<u32, &'static str> {
-            Ok(<XDNSRegistry<T>>::get(chain_id)
-                .ok_or("XDNSRegistry does not contain given chain id")?
-                .parachain
-                .ok_or("Xdns record doesn't have any Parachain data configured")?
-                .id)
+        fn get_gateway_para_id(chain_id: &ChainId) -> Result<u32, DispatchError> {
+            if !<XDNSRegistry<T>>::contains_key(chain_id) {
+                return Err(Error::<T>::XdnsRecordNotFound.into())
+            }
+
+            match <XDNSRegistry<T>>::get(chain_id).unwrap().parachain {
```

Comment:
same as above, also check rest for panics

Reply:
oui, indeed